app/src/main/java/me/ccrama/redditslide/Reddit.java

```diff
@@ -761,197 +550,88 @@ public void doMainStuff() {
-                notificationChannel.setShowBadge(true);
-                notificationChannel.enableVibration(true);
-                notificationChannel.setLightColor(Palette.getColor(""));
-                if (notificationManager != null) {
-                    notificationManager.createNotificationChannel(notificationChannel);
-                }
-            }
-            {
-                String channelId = CHANNEL_MODMAIL;
-                int importance = NotificationManager.IMPORTANCE_HIGH;
-                NotificationChannel notificationChannel =
-                        new NotificationChannel(channelId, "Reddit modmail", importance);
-                notificationChannel.enableLights(true);
-                notificationChannel.setSound(null, null);
-                notificationChannel.setShowBadge(true);
-                notificationChannel.setLightColor(getResources().getColor(R.color.md_red_500));
-                if (notificationManager != null) {
-                    notificationManager.createNotificationChannel(notificationChannel);
-                }
-            }
-            {
-                String channelId = CHANNEL_MODMAIL_SOUND;
-                int importance = NotificationManager.IMPORTANCE_HIGH;
-                NotificationChannel notificationChannel =
-                        new NotificationChannel(channelId, "Reddit modmail", importance);
-                notificationChannel.enableLights(true);
-                notificationChannel.setShowBadge(true);
-                notificationChannel.enableVibration(true);
-                notificationChannel.setLightColor(getResources().getColor(R.color.md_red_500));
-                if (notificationManager != null) {
-                    notificationManager.createNotificationChannel(notificationChannel);
-                }
-            }
-            {
-                String channelId = CHANNEL_SUBCHECKING;
-                int importance = NotificationManager.IMPORTANCE_LOW;
+            for (Triple<String, String, Integer> notificationTriple : notificationTripleList) {
                 NotificationChannel notificationChannel =
-                        new NotificationChannel(channelId, "Submission post checking", importance);
+                        new NotificationChannel(notificationTriple.getLeft(),
+                                notificationTriple.getMiddle(), notificationTriple.getRight());
                 notificationChannel.enableLights(true);
-                notificationChannel.setSound(null, null);
-                notificationChannel.setLightColor(Palette.getColor(""));
-                notificationChannel.setShowBadge(false);
+                notificationChannel.setShowBadge(
+                        notificationTriple.getRight() == NotificationManager.IMPORTANCE_HIGH);
+                notificationChannel.setLightColor(
+                        notificationTriple.getLeft().contains("MODMAIL") ? getResources().getColor(

```

Comment:
I had to string-check the notification channel string since modmail seems to get its own color. This should not be a big deal as long as no one decides to change those strings.